src/column_family.rs

```diff
@@ -28,24 +29,64 @@ pub const DEFAULT_COLUMN_FAMILY_NAME: &str = "default";
 pub struct ColumnFamilyDescriptor {
     pub(crate) name: String,
     pub(crate) options: Options,
+    pub(crate) ttl: ColumnFamilyTtl,
 }
 
 impl ColumnFamilyDescriptor {
-    // Create a new column family descriptor with the specified name and options.
+    /// Create a new column family descriptor with the specified name and options.
+    /// *WARNING*:
+    /// Will use [`ColumnFamilyTtl::SameAsWholeDb`] as ttl.
     pub fn new<S>(name: S, options: Options) -> Self
     where
         S: Into<String>,
     {
         Self {
             name: name.into(),
             options,
+            ttl: ColumnFamilyTtl::SameAsWholeDb,
         }
     }
 
+    /// Create a new column family descriptor with the specified name, options, and ttl.
+    /// *WARNING*:
+    /// The ttl is applied only when DB is opened with [`crate::db::DB::open_with_ttl()`].
+    pub fn new_with_ttl<S>(name: S, options: Options, ttl: ColumnFamilyTtl) -> Self
+    where
+        S: Into<String>,
+    {
+        Self {
+            name: name.into(),
+            options,
+            ttl,
+        }
+    }
+
+    /// Sets ttl for the column family. It's applied only when DB is opened with
+    /// [`crate::db::DB::open_with_ttl()`]. Changing ttl after DB is opened has no effect.
+    pub fn set_ttl(&mut self, ttl: ColumnFamilyTtl) {
+        self.ttl = ttl;
+    }
+
     /// Get the name of the ColumnFamilyDescriptor.
     pub fn name(&self) -> &str {
         &self.name
     }
+
+    pub fn ttl(&self) -> ColumnFamilyTtl {
+        self.ttl
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+/// Specifies the TTL behavior for a column family.
+/// <https://github.com/facebook/rocksdb/blob/18cecb9c46b4c2a8b148659dac2fcab5a843d32b/include/rocksdb/utilities/db_ttl.h#L16-L46>
+pub enum ColumnFamilyTtl {
+    /// Will internally set TTL to -1 (disabled)
+    Disabled,
+    /// Will set ttl to the specified duration
+    Set(Duration),
+    /// Will use ttl specified at db open time
+    SameAsWholeDb,
 }
```

Comment:
A couple of suggestions here. Looks like it makes sense to add the `Default` trait to the enum.  And I would rename the `Set` variant to `Duration`. It's a rust idiomatic, I guess. 
```rust
#[derive(Debug, Default, Clone, Copy)]
 /// Specifies the TTL behavior for a column family.
 /// <https://github.com/facebook/rocksdb/blob/18cecb9c46b4c2a8b148659dac2fcab5a843d32b/include/rocksdb/utilities/db_ttl.h#L16-L46>
 pub enum ColumnFamilyTtl {
     /// Will internally set TTL to -1 (disabled)
     #[default]
     Disabled,
     /// Will set ttl to the specified duration
     Duration(Duration),
     /// Will use ttl specified at db open time
     SameAsDb,
 }
```

Reply:
By the way, I'm going to include this PR to #964 and I can do it by myself if you don't mind.

Reply:
Maybe InheritDbTtl?

Reply:
> Naming is hard :)
Sometimes it is 😄

But as for me, whether `SameAsWholeDb` or  `SameAsDb` is fine.